Library/Homebrew/software_spec.rb

```diff
@@ -192,7 +192,7 @@ def depends_on(spec)
   def uses_from_macos(deps, bounds = {})
     if deps.is_a?(Hash)
       bounds = deps.dup
-      deps = [bounds.shift].to_h
+      deps = [T.unsafe(bounds).shift].to_h
```

Comment:
Can we just add a `sig` here instead?

Reply:
LMK if you have ideas, but i think it gets gnarly with type parameters and union types (and I think it would need to be in an rbi file due to the former). I think it's something I'd prefer to leave to a future PR (I'm thinking of doing a sweep over the T.unsafe call sites at some point anyway).

Reply:
I think you can use multiple `sig`s.

Reply:
```ruby
sig { params(deps: String, bounds: T::Hash[String, T.any(Symbol, T::Array[Symbol])]).void }
sig { params(deps: T::Hash[String, T.any(Symbol, T::Array[Symbol])]).void }
def uses_from_macos(deps, bounds = {})
```

Actually, I think we don't even need the `bounds` parameter.

Reply:
Overloading a method (by providing multiple signatures for the method) is only [allowed](https://sorbet.org/docs/error-reference#5040) for methods defined in the Ruby standard library.

Also, all params must be present in a `sig`.

(As a favor to the PR author, it would be appreciated to validate suggested changes before publishing them. Again, I also worry that there is excessive focus on this line, given the scope of this PR, and it may be better to refine in a future PR.)

Reply:
> As a favor to the PR author, it would be appreciated to validate suggested changes before publishing them.

I usually do, however I reviewed this on mobile.